geoips/interfaces/module_based/output_checkers.py

```diff
@@ -0,0 +1,500 @@
+        output_products : list of str
+            List of strings of current output products,
+            to compare with products in compare_path
+        test_product_func : function, default=None
+            Alternative function to be used for testing output product
+
+            * Call signature must be:
+
+                * output_product, compare_product, goodcomps, badcomps, compare_strings
+
+            * Return must be:
+
+                * goodcomps, badcomps, compare_strings
+
+            * If None, use geoips.compare_outputs.test_product)
+
+        Returns
+        -------
+        int
+            Binary code: 0 if all comparisons were completed successfully.
+        """
+        badcomps = []
+        goodcomps = []
+        missingcomps = []
+        missingproducts = []
+        compare_strings = []
+        LOG.info(
+            "**************************************************************************"
+            "****"
+        )
+        LOG.info(
+            "**************************************************************************"
+            "****"
+        )
+        LOG.info("*** RUNNING COMPARISONS OF KNOWN OUTPUTS IN %s ***", compare_path)
+        LOG.info(
+            "**************************************************************************"
+            "****"
+        )
+        LOG.info(
+            "**************************************************************************"
+            "****"
+        )
+        LOG.info("")
+        from glob import glob
+        from geoips.plugins.modules.output_checkers import gz_checker
+        from os.path import basename, exists, isdir, isfile, join
+
+        compare_basenames = [basename(yy) for yy in glob(compare_path + "/*")]
+        final_output_products = []
+
+        for output_product in output_products:
+            if isdir(output_product):
+                # Skip 'diff' output directory
+                continue
+
+            LOG.info(
+                "**********************************************************************"
+                "****"
+            )
+            LOG.info("*** COMPARE  %s ***", basename(output_product))
+            LOG.info(
+                "**********************************************************************"
+                "****"
+            )
+
+            rezip = False
+            if gz_checker.correct_type(output_product):
+                rezip = True
+                output_product = gz_checker.gunzip_product(output_product)
+            if basename(output_product) in compare_basenames:
+                if test_product_func is None:
+                    test_product_func = self.test_products
+                goodcomps, badcomps, compare_strings = test_product_func(
+                    output_product,
+                    join(compare_path, basename(output_product)),
+                    goodcomps,
+                    badcomps,
+                    compare_strings,
+                )
```

Comment:
Should "test_product" just be another expected method on each plugin within this class?  So rather than passing in the "test_product_func" function, just call self.test_product ?